### DISCUSSION TITLE
Missing two key feature.

### DISCUSSION THREAD

**Discussion by m4rcelpl** [pocketbase/pocketbase#1519] [Q&A] [answered]
Concept of Pocketbase is very interesting. I would like to use it, in ma internal projects. But is missing two key feature or i don't know how to use it*.

1. One, not expire admin api key to communicate server-to-server. I can't even generate api key for my account from admin panel. 
2. Audit log. I know that SQLite don't support triggers, to write all changes to another audit table. But maybe we can achieve this in another way? I need some controlling toll logs, audit etc. 

* Yes a know that i can add everything using Pocketbase like module, but i don't know GO :(

**Comment by ganigeorgiev** [comment, marked as answer]
1. Admin API-keys/permanent-tokens were requested previously in https://github.com/pocketbase/pocketbase/discussions/342 and it is in my todo to research it (together with admin roles), but for now it is a low priority.

2. I'm not sure if we need this by default and it will be difficult to generalize. As you noted, you can implement it on your own by attaching to the [record event hooks](https://pocketbase.io/docs/event-hooks/#record-api-hooks). Option to bind external code with JS (goja) is planned for v0.13/v0.14.

> I know that SQLite don't support triggers...

That's not true. SQLite has triggers and this approach could be a good alternative if your audit log doesn't require advanced/custom business logic - https://www.sqlite.org/lang_createtrigger.html.

**Reply by m4rcelpl** [reply]
Thx for answered. I will wait for future update. 

You are right, SQLite had triggers, I missed that. But we can’t make custom queries from admin panel and set it. (I check demo only, correct me if I’m wrong)

**Reply by ganigeorgiev** [reply]
You can connect to your local db file located in `pb_data/data.db` using the official [sqlite3 cli tool](https://sqlite.org/cli.html) and execute the trigger from there OR run the trigger using a JS migration, aka. `./pocketbase migrate create 'my_new_migration'` (more information on this will be available soon).
